components/BarcodeScanner.vue

```diff
@@ -138,28 +155,51 @@ export default {
       cameraConstraints.zoom.step = step;
     }
 
+    async function emitScanResult({barcodeDetector, video}) {
+      const {signal} = abortController;
+      const barcodes = await detectBarcodes({barcodeDetector, video, signal});
+      const [result] = barcodes;
+      if(result?.format && result?.rawValue) {
```

Comment:
I don't think it's possible for `result` to be undefined/falsey now with our new design, i.e., `detectBarcodes` always returns 1 or more barcodes. So we can simplify this and just use the result, I'd think.